cmd/kubetoken/main.go

```diff
@@ -182,13 +209,25 @@ func chooseRole(roles []string) (string, error) {
 	return roles[n], nil
 }
 
-func filterRoles(roles []string, filter string) ([]string, error) {
+func filterRoles(roles []string, filter string, keyWordsList []string) ([]string, error) {
 	re, err := regexp.Compile(filter)
 	if err != nil {
 		return nil, err
 	}
 	for i := 0; i < len(roles); {
+		regexMatched := false
 		if re.MatchString(roles[i]) {
+			regexMatched = true
+		}
+		keyWordsMatched := true
+		for _, f := range keyWordsList {
+			if strings.Contains(roles[i], f) {
```

Comment:
Do we want to convert the role and keyword to lowercase?

Reply:
addressed in the last commit to do case insensitive match.